### DISCUSSION TITLE
Run DefectDojo locally on windows

### DISCUSSION THREAD

**Discussion by flmarkus** [DefectDojo/django-DefectDojo#9379] [Q&A]
Primarliy I want to share my experience on how to run DefectDojo from source on windows.

Main pitfalls:

- Make sure you checkout the codebase with LF line endings
Setup git config: `git config core.autocrlf input` and `git config core.eol lf`
To check current line endings: `git ls-files --eol`
Reset line endings for already checked out files: `git rm -rf --cached .` and `git reset --hard HEAD`

- Run compose commands from bash scripts directly
Build with `docker compose --profile mysql-rabbitmq --profile postgres-redis --env-file ./docker/environments/postgres-redis.env build`
Run with `docker compose --profile postgres-redis --env-file ./docker/environments/postgres-redis.env up --no-deps -d`

- Check admin password
`docker compose logs initializer | Select-String -Pattern "Admin password:"`

My questions:

- Is windows as developement environment officially supported?
- If not, should linux be documented as developement requirement?
- If yes, should there be any documentation added to help getting started on a windows platform?
- Or is there already any kind of documentation for windows I have missed?

Thanks in advance for any kind of feedback!

**Comment by mtesauro** [comment]
Here's what I know:

> Is windows as developement environment officially supported?

Of the contributors, moderators, maintainer that I know, the vast majority of them are using either a Mac or Linux so a *nix like environment. I do know of one contributor who is using Windows, I believe they are heavily relying on WSL.

So, we don't officially support any particular OS for dev work, each has it's own wrinkles.

> If not, should linux be documented as developement requirement?

I'm the last one to answer this question since I only use Linux - I don't have a Windows or Mac so don't have direct experience on how closely those other OSes map to Linux in general or for DefectDojo work.

There is a very strong Linux background and history with running DefectDojo since  when it was first written, it was targeting "Debian" style Linux environments. Over the years, DefectDojo has grown to work on pretty much any Linux but I don't know of anyone running DefectDojo in a non-Dev way on anything but a Linux distro/docker/k8s - and docker/k8s are pretty "Linux-y"

> If yes, should there be any documentation added to help getting started on a windows platform?

I'm certainly not opposed to documentation that helps people work with DefectDojo. It should probably be either in the ["Community Contribs" repo](https://github.com/DefectDojo/Community-Contribs) where things tangential to DefectDojo live or perhaps a brief .md page at https://github.com/DefectDojo/django-DefectDojo/tree/master/readme-docs  I've got a couple of guides I've added there over the years like [this one](https://github.com/DefectDojo/Community-Contribs/tree/master/CentOS-7-deploy).

> Or is there already any kind of documentation for windows I have missed?

Not that I know of - hopefully others who use Windows can chime in on this.

HTH
